File: src/main/go/leetcode/lists/topinterview/LeetCode116.go
package topinterview

func connect(root *Node) *Node {
	if root == nil {
		return nil
	}
	cur := root
	for cur != nil {
		nextLevel := cur.Left
		next := cur
		for next != nil {
			if next.Left != nil {
				next.Left.Next = next.Right
			}
			if next.Right != nil && next.Next != nil {
				next.Right.Next = next.Next.Left
			}
			next = next.Next
		}
		cur = nextLevel
	}
	return root
}

File: java/src/leetcode/problems/LeetCode116.java
package leetcode.problems;

import common.*;

/**
 * @author wujingxinit@outlook.com
 * @date 2024/4/16 14:55
 */
public class LeetCode116 {

    public Node connect(Node root) {
        Node head = root;
        while (head != null) {
            Node dummy = new Node();
            Node cur = dummy;
            while (head != null) {
                if (head.left != null) {
                    cur.next = head.left;
                    cur = cur.next;
                }
                if (head.right != null) {
                    cur.next = head.right;
                    cur = cur.next;
                }
                head = head.next;
            }
            head = dummy.next;
        }
        return root;
    }
}
